src/web/app/components/question-responses/student-view-responses/student-view-responses.component.html

```diff
@@ -16,6 +16,35 @@
               <tm-single-response [responseDetails]="response.responseDetails" [questionDetails]="questionDetails" [isStudentPage]="true"></tm-single-response>
             </td>
           </tr>
+          <tr *ngIf="response.participantComment">
+            <td>
+              <tm-comment-row
+                              [mode]="CommentRowMode.EDIT"
+                              [isVisibilityOptionEnabled]="false"
+                              [isDisabled]="true"
+                              [shouldHideSavingButton]="true"
+                              [shouldHideClosingButton]="true"
+                              [shouldHideEditButton]="true"
+                              [shouldHideDeleteButton]="true"
+                              [isFeedbackParticipantComment]="true"
+                              [response]="response"
+                              [questionShowResponsesTo]="[]"
+                              [model]="
+                                {
+                                  commentEditFormModel: {
+                                    commentText: response.participantComment.commentText,
+                                    isUsingCustomVisibilities: false,
+                                    showCommentTo: response.participantComment.showCommentTo,
+                                    showGiverNameTo: response.participantComment.showGiverNameTo
+                                  },
+                                  originalComment: response.participantComment,
+                                  isEditing: false,
+                                  commentGiverName: response.participantComment.commentGiverName,
+                                  lastEditorName: response.participantComment.lastEditorName
+                                }"
```

Comment:
Is it possible to move this to the ts file instead? Genuine question, I'm not that familiar with Angular haha

Reply:
Gotcha will ask senior devs for input on this

Reply:
Searching for examples in code base is something a good option:

https://github.com/TEAMMATES/teammates/blob/0e168752753ffdd1f1a85e71c15878edb5ae9a1d/src/web/app/components/question-responses/grouped-responses/grouped-responses.component.html#L22-L30

Reply:
Thanks for the suggestion! However, I realise that the same function is ran multiple times whenever I do something on the page, like clicking on anything on the page. Most likely something to do with Angular's change detection mechanism.

Do you think it's worth optimizing this, like using a pure pipe instead of simply calling a function in the template?

Reply:
Do something on the page? Like?

Reply:
Yup just clicking anywhere on the page, clickable or not.

Reply:
Let's just be clear.

If you make a method like `transformParticipantCommentToCommandRowModel`, then you are saying that when you do something on the page, the method gets called multiple time.  How do you know that the method is ran multiple times (by adding `console.log`?)

Reply:
> If you make a method like `transformParticipantCommentToCommandRowModel`, then you are saying that when you do something on the page, the method gets called multiple time. 

Yes. Actually the current `transformParticipantCommentToCommandRowModel` function on the `master` branch also suffers from the same problem.

>  How do you know that the method is ran multiple times (by adding `console.log`?)

To reproduce:

1. Add a `console.log("calling transform");` to the current `transformParticipantCommentToCommandRowModel` function
2. Navigate to a page that uses `GroupedResponsesComponent such as the instructor session result page.
3. Ensure that it has participant comments **and** the responses are grouped by GRQ or RGQ
4. Click anywhere on the page.

![Untitled_ May 27, 2020 1_42 PM](https://user-images.githubusercontent.com/32454748/82982204-3f52a580-a020-11ea-9527-5b6407146450.gif)

Reply:
https://medium.com/showpad-engineering/why-you-should-never-use-function-calls-in-angular-template-expressions-e1a50f9c0496

Good catching 💯 

So if we construct the object in the template file, can we avoid this?

Reply:
That's what I hoped... But I can't find any mention of this strategy anywhere, and I'm not sure on how to test it as well.

Personally think that pure pipe is better, since we can facilitate code reuse while resolving this issue, as long as the input `participantComment: CommentOutput` is immutable.

Reply:
I think we can try pipe. You can add one in comment-box module.

Reply:
Got it. Will do.